src/crypto/slow-hash.c

```diff
@@ -515,8 +515,11 @@ void slow_hash_free_state(void)
  * @param length the length in bytes of the data
  * @param hash a pointer to a buffer in which the final 256 bit hash will be stored
  */
+void cn_slow_hash(const void *data, size_t length, char *hash) {
+    cn_slow_hash_pre(data,length,hash,0);
+}
 
-void cn_slow_hash(const void *data, size_t length, char *hash)
+void cn_slow_hash_pre(const void *data, size_t length, char *hash, bool prehashed)
```

Comment:
I cannot find anyone call this function with prehashed set to true. There is another function, `generate_chacha_key`, which takes a default argument that is always false. Please drop this variant entirely unless there is an immediate need for this function. It just confuses a critical function in the codebase.

Reply:
Call sequence is:

    bool  DeviceLedger::generate_chacha_key(const cryptonote::account_keys &keys, crypto::chacha_key &key)
        |->  inline void generate_chacha_key(const void *data, size_t size, chacha_key& key, bool prehashed)  WITH crypto::generate_chacha_key(&prekey[0], sizeof(prekey), key, true);
            |->  void cn_slow_hash_pre(const void *data, size_t length, char *hash, bool pre) WITH  crypto::cn_slow_hash_pre(data, size, pwd_hash.data(), prehashed);